cpp/include/cuml/cluster/kmeans_mg.hpp

```diff
@@ -16,10 +16,13 @@
 
 #pragma once
 
-#include "kmeans.hpp"
+namespace raft {
+class handle_t;
+}
```

Comment:
It seems that we might end up repeating this a lot of times in the code, it's few lines so not the biggest deal but still wondering if it's the best solution. I guess creating a super small .h for this is overkill, but wanted to hear your thoughts.

Reply:
I think it might be best to do it in the PR with the forward declarations so that it fits with the work in it (or another PR if you think is best), so will go ahead and merge. Thanks!